lectures/04_lecture/data_pipelines.ipynb

```diff
@@ -0,0 +1,1573 @@
+       "    <tr>\n",
+       "      <th>RandomForestClassifier(class_weight='balanced', max_depth=80)</th>\n",
+       "      <td>0.899820</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>MLPClassifier(hidden_layer_sizes=2)</th>\n",
+       "      <td>0.813091</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>MLPClassifier(hidden_layer_sizes=6)</th>\n",
+       "      <td>0.813091</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>MLPClassifier(hidden_layer_sizes=10)</th>\n",
+       "      <td>0.865318</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>MLPClassifier(hidden_layer_sizes=20)</th>\n",
+       "      <td>0.872352</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>MLPClassifier(hidden_layer_sizes=40)</th>\n",
+       "      <td>0.876567</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "                                                       score\n",
+       "LogisticRegression(multi_class='multinomial')       0.868228\n",
+       "RandomForestClassifier(class_weight='balanced',...  0.814415\n",
+       "RandomForestClassifier(class_weight='balanced',...  0.865303\n",
+       "RandomForestClassifier(class_weight='balanced',...  0.898822\n",
+       "RandomForestClassifier(class_weight='balanced',...  0.894330\n",
+       "RandomForestClassifier(class_weight='balanced',...  0.899820\n",
+       "MLPClassifier(hidden_layer_sizes=2)                 0.813091\n",
+       "MLPClassifier(hidden_layer_sizes=6)                 0.813091\n",
+       "MLPClassifier(hidden_layer_sizes=10)                0.865318\n",
+       "MLPClassifier(hidden_layer_sizes=20)                0.872352\n",
+       "MLPClassifier(hidden_layer_sizes=40)                0.876567"
+      ]
+     },
+     "execution_count": 21,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from sklearn.metrics import f1_score\n",
+    "\n",
+    "results = {}\n",
+    "for model in models_to_run:\n",
+    "\n",
+    "    # create instance of the class, feed the model to be tested\n",
+    "    clf = make_pipeline(categorical_cols, numerical_cols, model)\n",
+    "    clf.fit(X_train_wine, y_train_wine)\n",
+    "\n",
+    "    # store results in a dictionary \n",
+    "    results[str(model)] = f1_score(y_test_wine, clf.predict(X_test_wine), average='weighted')\n",
+    "pd.DataFrame.from_dict(results, orient='index', columns=['score'])\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Unfortunately the parameter did not change our model for the better. This can ofcourse always happen in Machine Learning, which is sometimes more of an art than a science. Several other methods to combat class imbalance could be checked for (oversampling, smote...), but an increase in performance is never guaruanteed. "
```

Comment:
```suggestion
    "Unfortunately the parameter did not change our model for the better. This can of course always happen in Machine Learning, which is sometimes more of an art than a science. Several other methods to combat class imbalance could be checked for (oversampling, smote...), but an increase in performance is never guaranteed. "
```

Reply:
I would also recommend to add a link to the classes you mention.